Framework/DataHandling/src/LoadMuonNexusV2Helper.cpp

```diff
@@ -188,6 +203,66 @@ getLoadedDetectors(const DataObjects::Workspace2D_sptr &localWorkspace) {
   return loadedDetectors;
 }
 
+MuonNexus::SampleInformation
+loadSampleInformationFromNexus(const NeXus::NXEntry &entry) {
+  auto runSample = entry.openNXGroup(NeXusEntry::SAMPLE);
+  MuonNexus::SampleInformation sampleInformation;
+  try {
+    sampleInformation.magneticField =
+        runSample.getFloat(NeXusEntry::MAGNETICFIELD);
+    sampleInformation.temperature = runSample.getFloat(NeXusEntry::TEMPERATURE);
+  } catch (std::runtime_error) {
+    throw std::runtime_error("Could not load sample information (temperature "
+                             "and magnetic field) from nexus entry");
+  }
+  return sampleInformation;
+}
+/**
+ * Loads default detector grouping, if this isn't present
+ * return dummy grouping
+ * @returns :: Grouping table
+ */
+Workspace_sptr loadDefaultDetectorGrouping(
+    const DataObjects::Workspace2D_sptr &localWorkspace) {
```

Comment:
Ditto about dropping shared pointers for all of these if they're called by algs which will have their own copy of the shared pointer (like `LoadMuonNexus`)